kotlin/lib/src/main/kotlin/truelayer/signing/Exceptions.kt

```diff
@@ -0,0 +1,32 @@
+package truelayer.signing
+
+class InvalidKeyException private constructor(message: String, cause: Throwable) : Exception(message, cause) {
+    companion object {
+        internal fun <T> evaluate(block: () -> T): Result<T> = runCatching {
+            block()
+        }.mapThrowable { e -> InvalidKeyException("Invalid Key: ${e.message}", e) }
+    }
+}
+
+class InvalidSignatureException : Exception {
```

Comment:
In C# I called this simply `SignatureException` so could use that here to for "consistency" but its not a big deal.

Reply:
The other one similarly could just be `KeyException` if you read "exception" to mean error, which is does, then it is enough.